### ISSUE TITLE
(phi-deidentifier) Add endpoint that takes as input information to get note from a Data Node

### ISSUE THREAD

**Issue by tschaffter** [nlpsandbox/nlpsandbox-schemas#118]
The current endpoint that we are developing takes as input a clinical note. This is useful, among other, to demonstrate the PHI Deidentifier as a user can send a clinical note and get its de identifier version.

In a production settings, it would be interesting to give users access to deidentied notes without requiring the users to possess the clear note. Instead the user would provide as input the ID or resource name of the note. It's then up to the deidentifier to retrieve the node in a Data Node and return the result to the user.

**Comment by cascadianblue** [comment]
Would this (presumably non-privileged) user be able to set the de-identification strategies (e.g. masking char, redaction, annotation type, date offset)?

**Comment by tschaffter** [comment]
Here are some notes. This ticket is currently assigned to the next sprint (21.1) so we can further discuss before starting the implementation.

Motivation: illustrate a way the deidentifier could be used in practices where researchers provide the public ID of a patient - also called hashed ID often - and get back the deidentifier notes without other human intervention. The user could be a researcher at the hospital that setup the deidentifier or anyone. Alternatively, the deid clinical notes returned can come from a DB that include pre-approved clinical notes generated previously. For now this use case is just a demo/proof of concept so we don't need to think to all the edge cases.

## Task

Add an endpoint with the following specifications

Input:
  - public ID of a patient
  - preferences: the deid could allow some preferences to be specified, e.g. whether dates should be offsetted or replaced by masking character, but there would be no way to disable the deid of this type of PHI in the current use case.
Output: list of deid clinical notes

Action:
1) retrieve the notes associated to the public patient ID given
2) deidentify them live OR retrieve deid notes from a DB
3) return the deidentified notes

**Comment by tschaffter** [comment]
## Update

This ticket assumes that an instance of the data node and PHI deidentifier are deployed and can communicate. This could be achieved by putting together a NLP Sandbox Healthcare API. :) This is not a priority for now.
